scripts/deployer.ts

```diff
@@ -0,0 +1,197 @@
+import { ethers } from 'hardhat';
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+import { DeployFunction } from 'hardhat-deploy/types';
+import 'hardhat-deploy-ethers';
+import hre from 'hardhat';
+import * as fs from 'fs';
+import { ContractReceipt, ContractTransaction } from 'ethers';
+import mainnetData from '../mainnet_deployed.json';
+import testnetData from '../testnet_deployed.json';
+import dummyData from '../example_testnet_deployed.json';
+import { getCurrentTimestamp } from 'hardhat/internal/hardhat-network/provider/utils/getCurrentTimestamp';
+
+let deployedData = dummyData;
+let fileName = 'example_testnet_deployed.json';
+// refactor:
+//  mainnet_deployed.json
+//  testnet_deployed.json
+//  local_deployed.json
+
+// URL
+const mainnetURL = 'https://gnosisscan.io/address/';
+const testnetURL = 'https://goerli.etherscan.io/address/';
+
+// Chain Vendor
+const blockChainVendor = hre.network.name;
+const networkID = hre.network.config.chainId;
+
+// all contract addresses will be set here
+let factory = '';
+let staking = '';
+let postage = '';
+let oracle = '';
+let redistribution = '';
+
+// all block numbers will be set here
+let factoryBlock = 0;
+let stakingBlock = 0;
+let postageBlock = 0;
+let oracleBlock = 0;
+let redistributionBlock = 0;
+
+async function main() {
+  if (blockChainVendor != 'testnet' && blockChainVendor != 'mainnet') {
+    const TestToken = await hre.ethers.getContractFactory('TestToken');
+    const testToken = await TestToken.deploy();
+    await testToken.deployed();
+    factory = testToken.address;
+    factoryBlock = testToken.deployTransaction.blockNumber as number;
+  } else if (blockChainVendor == 'testnet') {
+    const deployed = await JSON.parse(JSON.stringify(testnetData).toString());
```

Comment:
Why do you use this construct? IMHO you should not need it...

Generally, this is used to make a deep copy of an object, but it is rarely used and usually signals some bad approach.